aptos-move/aptos-aggregator/src/aggregator_extension.rs

```diff
@@ -393,39 +412,103 @@ mod test {
         let mut aggregator_data = AggregatorData::default();
         aggregator_data.create_new_aggregator(aggregator_id_for_test(200), 200);
 
-        let aggregator = aggregator_data.get_aggregator(aggregator_id_for_test(200), 200);
+        let aggregator =
+            aggregator_data.get_aggregator(aggregator_id_for_test(200), 200, &*TEST_RESOLVER, true);
         assert_ok!(aggregator.add(100));
         assert_ok!(aggregator.read_and_materialize(&*TEST_RESOLVER, &aggregator_id_for_test(200)));
         assert_eq!(aggregator.value, 100);
     }
 
+    #[test]
+    fn test_materialize_known_aggregator_disabled() {
+        let mut aggregator_data = AggregatorData::default();
+        aggregator_data.create_new_aggregator(aggregator_id_for_test(200), 200);
+        let aggregator =
+            aggregator_data.get_aggregator(aggregator_id_for_test(200), 200, &*TEST_RESOLVER, true);
+        assert_ok!(aggregator.add(100));
+        assert_ok!(aggregator.read_and_materialize(&*TEST_RESOLVER, &aggregator_id_for_test(200)));
+
+        let aggregator = aggregator_data.get_aggregator(
+            aggregator_id_for_test(200),
+            200,
+            &*TEST_RESOLVER,
+            false,
+        );
+        assert_eq!(aggregator.state, AggregatorState::Data);
+        assert_eq!(aggregator.value, 100);
+        assert_ok!(aggregator.add(50));
+        assert_eq!(aggregator.state, AggregatorState::Data);
+        assert_eq!(aggregator.value, 150);
+    }
+
     #[test]
     fn test_materialize_overflow() {
         let mut aggregator_data = AggregatorData::default();
 
         // +0 to +400 satisfies <= 600 and is ok, but materialization fails
         // with 300 + 400 > 600!
-        let aggregator = aggregator_data.get_aggregator(aggregator_id_for_test(600), 600);
+        let aggregator =
+            aggregator_data.get_aggregator(aggregator_id_for_test(600), 600, &*TEST_RESOLVER, true);
         assert_ok!(aggregator.add(400));
         assert_err!(aggregator.read_and_materialize(&*TEST_RESOLVER, &aggregator_id_for_test(600)));
     }
 
+    #[test]
+    fn test_materialize_overflow_aggregator_disabled() {
+        let mut aggregator_data = AggregatorData::default();
+        aggregator_data.create_new_aggregator(aggregator_id_for_test(200), 600);
```

Comment:
Btw, this function (`create_new_aggregator`) creates aggregator in data state, so calling get on it always returns data. Maybe let's add some tests where aggregator is not known so we call `get_aggregator` with the right flag only?

Reply:
I added another test case that doesn't invoke `create_new_aggregator` in the beginning. Please let me know if it works.

Reply:
Yep, this is a great thing to test, and we can check "expect" or whatever behavior happens here depending on the earlier thread.